Unicode locale

kristomu opened the pull request:
This should fix the AltGr problem of my last PR.

kristomu committed:
Fix regression that ate Alt+arrow key inputs.

SDL2's KMOD_MODE never did check for AltGr; I'm not sure what it's for.

kristomu committed:
Merge branch 'unicode_locale' of github.com:kristomu/kevedit into unicode_locale

kristomu committed:
Minor fix: unicode.c was mixing spaces and tabs.

kristomu committed:
Fix regression where shift wouldn't register in OOP editing.

I didn't notice that "shift" was a global variable.

cknave commented on src/display/display_sdl.c:
```diff
@@ -578,6 +578,23 @@ static void display_sdl_fullscreen()
+   produces. */
+static int has_unicode_event_queued()
+{
+	SDL_Event outevent;
+	int pending_events = SDL_PeepEvents(&outevent,
+		1, SDL_PEEKEVENT, SDL_FIRSTEVENT, SDL_LASTEVENT);
+
+	if (pending_events < 1)
+		return 0;
+
+	return outevent.type == SDL_TEXTINPUT &&
+		!is_ascii_key(outevent.text.text[0]);
```
If the user is moving the mouse or their CPU gets bogged down, there could be more than 1 event queued. We can deal with this using `SDL_PeepEvents` filtering and a larger buffer. Something like:

```suggestion
	const int MAX_EVENTS = 10;
	SDL_Event outevent[MAX_EVENTS];
	int pending_events = SDL_PeepEvents(outevent,
		MAX_EVENTS, SDL_PEEKEVENT, SDL_TEXTINPUT, SDL_TEXTINPUT);
	int i;
	for (i = 0; i < pending_events; i++) {
		if (!is_ascii_key(outevent[i].text.text[0])) {
			return true;
		}
	}
	return false;
```

cknave commented:
Aside from that, I tested this branch with the compose key on linux and it works great!

kristomu committed:
Poll more than one event in has_unicode_event_queued; minor cleanup.

The minor cleanup in unicode is mainly to use a simpler function for
unicode decoding.

kristomu committed:
Clean up has_unicode_event_queued.
